job/job.go

```diff
@@ -409,14 +412,31 @@ func (j *Job) DeleteFromDependentJobs(cache JobCache) error {
 	return nil
 }
 
+// Runs the on failure job, if it exists. Does not lock the job - it is up to you to do this
```

Comment:
We should lock the reading of the OnFailureJob

Reply:
I believe that is already taken care of in Job.Run, or did you mean something else?